lib/ClassUtils.lua

```diff
@@ -129,13 +129,23 @@ function ClassUtils.applySwitchStrategyForEnum(enum, enumValue, strategies, ...)
 	return strategies[enumValue](...)
 end
 
-function ClassUtils.makeSymbolEnum(keys)
-	return TableUtils.map(
-		ClassUtils.makeEnum(keys),
-		function(key)
-			return ClassUtils.makeSymbol(key)
+function ClassUtils.makeFinal(object)
+	local backend = getmetatable(object)
+	local proxy = {
+		__index = function(t, key)
```

Comment:
Will we ever iterate over these objects using `ipairs` or `pairs`. Apparently that won't work: http://lua-users.org/wiki/ReadOnlyTables. Worth testing?